gateway/src/main/java/io/camunda/zeebe/gateway/admin/backup/BackupRequestHandler.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.gateway.admin.backup;
+
+import io.camunda.zeebe.gateway.cmd.NoTopologyAvailableException;
+import io.camunda.zeebe.gateway.impl.broker.BrokerClient;
+import io.camunda.zeebe.gateway.impl.broker.cluster.BrokerClusterState;
+import io.camunda.zeebe.gateway.impl.broker.cluster.BrokerTopologyManager;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.IntStream;
+
+public class BackupRequestHandler implements BackupApi {
+
+  final BrokerClient brokerClient;
+  final BrokerTopologyManager topologyManager;
+
+  public BackupRequestHandler(final BrokerClient brokerClient) {
+    this.brokerClient = brokerClient;
+    topologyManager = brokerClient.getTopologyManager();
+  }
+
+  @Override
+  public CompletableFuture<Long> takeBackup(final long backupId) {
+    final BrokerClusterState topology = topologyManager.getTopology();
+    if (topology == null) {
+      return CompletableFuture.failedFuture(new NoTopologyAvailableException());
+    }
+
+    final int partitionsCount = topology.getPartitionsCount();
```

Comment:
:wrench: This tripped me up a bit, I was wondering why not just use `topology.getPartitions()`, but then realized it's to deal with incomplete topologies. Maybe we can add a comment here why we're specifically going via the partition count? 

:upside_down_face: It also feels somewhat brittle to count 1 to partition count, and use that as the ID. I'm wondering if it might not make more sense to do something like `IntStream.range(0, partitionsCount).map(topology::getPartition)`, to decouple that. Just a minor nit, feel free to ignore, I'm not even convinced by myself :smile:

Reply:
Yes. It is brittle because we are assuming how partition ids are generated. 
We cannot use `topology::getPartition` because it returns a valid id only if it has full topology. 

Alternate: we can check if `topology.getPartitionCount() == topology.getPartitions().size()`. And then use `topology.getPartitions()`.

Reply:
Assuming the error you get when the partition is unknown is alright, then I think how you did it is fine, maybe we can add a comment since the why is not immediately clear I think.

Reply:
I updated it to check if topology.getPartitionCount() == topology.getPartitions().size(). And then use topology.getPartitions(). This is safer, as we are not assuming that partitionIds starts from 1.